src/main/java/com/nanalysis/sharepoint/SharepointClient.java

```diff
@@ -155,6 +157,52 @@ public void uploadFile(String folder, String filename, byte[] data) throws IOExc
         checkForError(response);
     }
 
+    public void uploadBigFile(String folder, String filename, long size, InputStream input, Consumer<Double> progressCallback)
+            throws IOException, InterruptedException {
+        // create small file first, it will be overwritten later
+        uploadFile(folder, filename, "PLACEHOLDER".getBytes());
+
+        String uid = UUID.randomUUID().toString();
+        byte[] buffer = new byte[10*1024*1024]; //10MB
+        int nread;
+        int offset = 0;
```

Comment:
what is nread?

Reply:
Oh ok, number of bytes read, which depends on the buffer. Always the buffer size until the last chunk.